sdc/extensions/indexes/range_index_ext.py

```diff
@@ -0,0 +1,366 @@
+
+        return range_index._getvalue()
+
+    ret_typ = RangeIndexType(is_named)
+    sig = signature(ret_typ, data, name)
+    return sig, codegen
+
+
+@sdc_overload(pd.RangeIndex)
+def pd_range_index_overload(start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+
+    _func_name = 'pd.RangeIndex().'
+    ty_checker = TypeChecker(_func_name)
+
+    if not (isinstance(copy, types.Omitted) or copy is False):
+        raise SDCLimitation(f"{_func_name} Unsupported parameter. Given 'copy': {copy}")
+
+    if not (isinstance(copy, types.Omitted) or fastpath is None):
+        raise SDCLimitation(f"{_func_name} Unsupported parameter. Given 'fastpath': {fastpath}")
+
+    dtype_is_np_int64 = dtype is types.NumberClass(types.int64)
+    if not _check_dtype_param_type(dtype):
+        ty_checker.raise_exc(dtype, 'int64 dtype', 'dtype')
+
+    # TODO: support ensure_python_int from pandas.core.dtype.common to handle integers as float params
+    if not (isinstance(start, (types.NoneType, types.Omitted, types.Integer)) or start is None):
+        ty_checker.raise_exc(start, 'number or none', 'start')
+    if not (isinstance(stop, (types.NoneType, types.Omitted, types.Integer)) or stop is None):
+        ty_checker.raise_exc(stop, 'number or none', 'stop')
+    if not (isinstance(step, (types.NoneType, types.Omitted, types.Integer)) or step is None):
+        ty_checker.raise_exc(step, 'number or none', 'step')
+
+    if not (isinstance(name, (types.NoneType, types.Omitted, types.StringLiteral, types.UnicodeType)) or name is None):
+        ty_checker.raise_exc(name, 'string or none', 'name')
+
+    if ((isinstance(start, (types.NoneType, types.Omitted)) or start is None)
+            and (isinstance(stop, (types.NoneType, types.Omitted)) or stop is None)
+            and (isinstance(step, (types.NoneType, types.Omitted)) or step is None)):
+        def pd_range_index_ctor_dummy_impl(
+                start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+            raise TypeError("RangeIndex(...) must be called with integers")
+
+        return pd_range_index_ctor_dummy_impl
+
+    def pd_range_index_ctor_impl(start=None, stop=None, step=None, dtype=None, copy=False, name=None, fastpath=None):
+
+        if not (dtype is None
+                or dtype == 'int64'
+                or dtype_is_np_int64):
+            raise TypeError("Invalid to pass a non-int64 dtype to RangeIndex")
+
+        _start = types.int64(start) if start is not None else types.int64(0)
+
+        if stop is None:
+            _start, _stop = types.int64(0), types.int64(start)
+        else:
+            _stop = types.int64(stop)
+
+        _step = types.int64(step) if step is not None else types.int64(1)
+        if _step == 0:
+            raise ValueError("Step must not be zero")
+
+        if _start == 0 and _stop == 0 and _step == 0:
+            raise TypeError("RangeIndex(...) must be called with integers")
```

Comment:
Is this correct error message? We are checking `_start`, `_stop` and `_step` to be non-zero. But reports what it must be integers

Reply:
Thanks! I incorrectly merged two different errors into one.